lib/http-bridge/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -306,25 +312,77 @@ expectEventually
 expectEventually ctx getter target (_, res) = case res of
     Left e -> wantedSuccessButError e
     Right s -> loopUntilRestore (s ^. walletId)
-
   where
     loopUntilRestore :: (MonadIO m, MonadCatch m) => Text -> m ()
     loopUntilRestore wid = do
         r <- request @ApiWallet ctx ("GET", "v2/wallets/" <> wid) Default Empty
         let target' = getFromResponse getter r
         unless (target' >= target) $ loopUntilRestore wid
 
+expectEventually'
+    :: (MonadIO m, MonadCatch m, MonadFail m, Ord a)
+    => Context
+    -> Lens' ApiWallet a
+    -> a
+    -> ApiWallet
+    -> m ()
+expectEventually' ctx target value wallet = do
```

Comment:
I am not fond of this one :no_mouth: ... It's rather cryptic, especially without any comment :grimacing:

Reply:
Comment added.:sweat_smile: